Universal FE Randomizer/src/random/gba/randomizer/shuffling/CharacterShuffler.java

```diff
@@ -0,0 +1,325 @@
+			ClassDataLoader classData, CharacterShufflingOptions options, ItemAssignmentOptions inventoryOptions, ItemDataLoader itemData) {
+		// Don't include playable characters into the ones that could be replaced, 
+		// as most files probably won't be played enough to unlock those anyway.
+		List<GBAFECharacterData> characterPool = new ArrayList<GBAFECharacterData>(
+				characterData.canonicalPlayableCharacters(false));
+
+		// Shuffle in character from the other games
+		if (options.getIncludedShuffles().size() != 0) {
+			PoolDistributor<GBACrossGameData> distributor = new PoolDistributor<GBACrossGameData>();
+			for (String includedCharacters : options.getIncludedShuffles()) {
+				distributor.addAll(CharacterImporter.importCharacterDataFromFiles(includedCharacters));
+			}
+			for (GBAFECharacterData slot : characterPool) {
+				// Determine if the current character should be replaced
+				if (options.getChance() < rng.nextInt(1, 100)) {
+					continue;
+				}
+
+				// The character should be replaced, get a random character to shuffle in
+				GBACrossGameData crossGameData = distributor.getRandomItem(rng, true);
+				if (crossGameData == null) {
+					// If no more character to find, then stop
+					return;
+				}
+				
+				DebugPrinter.log(DebugPrinter.Key.GBA_CHARACTER_SHUFFLING, String.format("Shuffling Character %s into Slot %d, which was originally %s", crossGameData.name, slot.getID(), slot.displayString()));
+
+				// (a) Get a valid class in the target game for what was configured for the character
+				int targetClassId =  GBACrossGameData.getEquivalentClass(type, crossGameData).getID();
+
+				if (targetClassId == 0) {
+					DebugPrinter.error(DebugPrinter.Key.GBA_CHARACTER_SHUFFLING, String.format("Couldn't find an applicable class for character %s class %s%n", crossGameData.name,
+							crossGameData.characterClass));
+					continue;
+				}
+				
+				// (b) initialize some variables needed later
+				GBAFEClassData targetClass = classData.classForID(targetClassId);
+				GBAFEClassData sourceClass = classData.classForID(slot.getClassID());
+				int slotLevel = slot.getLevel();
+
+				// (c) Insert the portrait for the current character into the rom and repoint the Portrait Data
+				try {
+					changePortrait(slot, portraitData, type, crossGameData, freeSpace, fileHandler);
+				} catch (Exception e) {
+					e.printStackTrace();
+					continue;
+				}
+				
+				// (d) Update name and Description
+				updateName(textData, portraitData, slot, crossGameData);
+				
+				// Give an option to not change the description to help with keeping track of which character is which
+				if (options.shouldChangeDescription()) {
+					// [0x1] = new line 
+					// [X] = end of text segment?
+					textData.setStringAtIndex(slot.getDescriptionIndex(),String.format("%s[0x1]%s[X]", crossGameData.description1, crossGameData.description2));
```

Comment:
Can this potentially add an extra [0x1] if you only have a one line description? Or did you make sure every character has a 2 line description?

Reply:
Every character that is pre-provided (i.e. fe6,7,8) has two lines, obviously there is the ability for the user to add their own custom files, which would certainly make this a possibility.

I did try it just now, with the Ilyana from the PR, and I don't see an immediate issue:
![2023-03-26 09_24_23-VBA-RR v23 6 svn480-LRC4 100%](https://user-images.githubusercontent.com/41942612/227761537-6446496f-3753-4183-a03c-4108b6f017e1.png)

Reply:
Ah, if there's no issues, then that's fine. Just wanted to make sure this wouldn't be a problem. Admittedly I don't remember all of the nuances of 0x1.